src/main/kotlin/ee/carlrobert/codegpt/codecompletions/InfillPromptTemplate.kt

```diff
@@ -1,42 +1,97 @@
+        override fun buildPrompt(infillDetails: InfillRequestDetails): String {
+            // see https://github.com/QwenLM/CodeQwen1.5?tab=readme-ov-file#2-file-level-code-completion-fill-in-the-middle
+            val infillPrompt =
+                "<fim_prefix>${infillDetails.prefix}<fim_suffix>${infillDetails.suffix}<fim_middle>"
+            return if (infillDetails.context == null || infillDetails.context.contextFiles.isNullOrEmpty()) {
+                infillPrompt
+            } else {
+                "<reponame>${infillDetails.context.repoName}\n" +
+                        infillDetails.context.contextFiles.map {
+                            "<file_sep>${it.path} \n" +
+                                    it.readText()
+                        }.joinToString("") { it + "\n" } +
+                        "<file_sep>${infillDetails.context.file.path} \n" +
+                        infillPrompt
+            }
         }
     },
     STABILITY("Stability AI", listOf("<|endoftext|>")) {
-        override fun buildPrompt(prefix: String, suffix: String): String {
-            return "<fim_prefix>$prefix<fim_suffix>$suffix<fim_middle>"
+        override fun buildPrompt(infillDetails: InfillRequestDetails): String {
+            return "<fim_prefix>${infillDetails.prefix}<fim_suffix>${infillDetails.suffix}<fim_middle>"
         }
     },
     DEEPSEEK_CODER("DeepSeek Coder", listOf("<|EOT|>")) {
-        override fun buildPrompt(prefix: String, suffix: String): String {
-            return "<｜fim▁begin｜>$prefix<｜fim▁hole｜>$suffix<｜fim▁end｜>"
+        override fun buildPrompt(infillDetails: InfillRequestDetails): String {
+            // see https://github.com/deepseek-ai/DeepSeek-Coder?tab=readme-ov-file#2-code-insertion
+            val infillPrompt =
+                "<｜fim▁begin｜>${infillDetails.prefix}<｜fim▁hole｜>${infillDetails.suffix}<｜fim▁end｜>"
+            return if (infillDetails.context == null || infillDetails.context.contextFiles.isNullOrEmpty()) {
+                infillPrompt
+            } else {
+                infillDetails.context.contextFiles.map { "#${it.path}\n" + it.readText() }
+                    .joinToString("") { it + "\n" } +
+                        "#${infillDetails.context.file.path}\n" +
+                        infillPrompt
+            }
+        }
+    },
+    STAR_CODER("StarCoder2", listOf("<|endoftext|>")) {
+        override fun buildPrompt(infillDetails: InfillRequestDetails): String {
+            // see https://huggingface.co/spaces/bigcode/bigcode-playground/blob/main/app.py
+            val infillPrompt =
+                "<fim_prefix>${infillDetails.prefix} <fim_suffix> ${infillDetails.suffix}<fim_middle>"
+            return if (infillDetails.context == null || infillDetails.context.contextFiles.isNullOrEmpty()) {
+                infillPrompt
+            } else {
+                "<reponame>${infillDetails.context.repoName}" +
```

Comment:
This is really cool! I didn't know it was possible to give in additional knowledge like this 👍